realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -453,4 +474,23 @@ private static int storeColumnIndices(ColumnIndices[] array, ColumnIndices colum
         array[candidateIndex] = columnIndices;
         return candidateIndex;
     }
+
+    public RealmConfiguration getConfiguration() {
+        return configuration;
+    }
+
+    public ColumnIndices[] getTypedColumnIndicesArray() {
+        return typedColumnIndicesArray;
+    }
+
+    /**
+     * @return the total global ref count.
+     */
+    private int getTotalGlobalRefCount() {
+        int totalRefCount = 0;
+        for (RealmCacheType type : RealmCacheType.values()) {
```

Comment:
`refAndCountMap` is an `EnumMap`, and
> EnumMap is implemented with a pair of arrays of the same length as the number of enum entries.

So `EnumMap.values()` just return the values array without creating anything:
```
    public Collection<V> values() {
        Collection<V> vs = values;
        if (vs != null)
            return vs;
        else
            return values = new Values();
    }
```

I just knew this, interesting. I guess no need to change this.